src/http/server.ts

```diff
@@ -464,6 +495,50 @@ export class Server {
     }
   }
 
+  /**
+   * Handle HTTP requests for virtual path assets.
+   *
+   * @param request - The request object.
+   *
+   * @returns The response as stringified JSON. This is only used for unit
+   * testing purposes.
+   */
+  public async handleHttpRequestForVirtualPathAsset(
+    request: Drash.Http.Request,
+  ): Promise<Drash.Interfaces.ResponseOutput> {
+    try {
+      await this.executeMiddlewareServerLevelBeforeRequest(request);
+
+      const response = this.getResponse(request);
+
+      // Set the response's Content-Type type header based on the request's URL.
+      // For example, if the request's URL is /public/style.css, then the
+      // Content-Type header should be set to text/css.
+      const mimeType = this.http_service.getMimeType(request.url, true);
+
+      if (mimeType) {
+        response.headers.set("Content-Type", mimeType);
+      }
```

Comment:
Wonder if this logic could be abstracted? i only wonder because i *think* we have the same logic somewhere else in this class - more of a query than a suggestion

Reply:
true. i'll see what can be abstracted out

Reply:
couldn't really abstract stuff, but i cleaned up the logic a bit: 16818b5

Reply:
wasn't really a problem tbh, lgtm tho